wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -518,12 +522,27 @@ instance ToSchema AddressValidity where
 instance Arbitrary AddressValidity where
   arbitrary = AddressValidity <$> arbitrary
 
+data AddressInfo = AddressInfo
+    { addrinfoWalletId     :: !WalletId
+    , addrinfoAccountIndex :: !AccountIndex
```

Comment:
I would be almost tempted to go with your suggestion of not creating a new type but expand the `WalletAddress` one. Another hard lesson we learnt from the V0 API is that there were too many piecemeal, ad-hoc types, which ended up complicating quite a lot working with the codebase.

If we go with the unified type route, then our API would look quite neat:

```
type API = Tags '["Addresses"] :>
      (    "addresses" :> WalletRequestParams
                       :> Summary "Returns a list of the addresses."
                       :> Get '[ValidJSON] (WalletResponse [WalletAddress])
      :<|> "addresses" :> ReqBody '[ValidJSON] NewAddress
                       :> Summary "Creates a new Address."
                       :> Post '[ValidJSON] (WalletResponse WalletAddress)
      :<|> "addresses" :> Capture "address" Text
                       :> Summary "Returns information about this address, if available"
                       :> Get '[ValidJSON] (WalletResponse WalletAddress)
      )
```

If means we have boiled down `/addresses` to the crux of the REST operations:

- list all resources
- create a new resource
- get a single resource

**Note**: I would (for now) exclude the `AccountIndex` from the returned type. In this crazy crypto world, the fewer info you share, the better, especially with mobile wallets at the horizon. Happy to explain in a call later 😉

Reply:
I'm in favor of this as well :smile: 

So, what fields do we want to add to WalletAddress?